astropy/tests/helper.py

```diff
@@ -211,7 +211,7 @@ def treat_deprecations_as_exceptions():
     warning state.
     """
     # First, totally reset the warning state
-    for module in six.itervalues(sys.modules):
+    for module in list(six.itervalues(sys.modules)):
```

Comment:
Could you add a small comment before, something like `The modules may change during the iteration - copy the current state to a list. [#5513]`